pkg/controller/utils/imageset/imageset.go

```diff
@@ -72,14 +85,49 @@ func GetImageSet(ctx context.Context, cli client.Client, v operator.ProductVaria
 	}
 
 	setName := getSetName(v)
+	vp := variantPrefix(v)
+	variantISExists := false
 
 	for _, is := range isl.Items {
+		if strings.HasPrefix(is.Name, vp) {
+			variantISExists = true
+		}
 		if is.Name == setName {
 			return &is, nil
 		}
 	}
 
-	return nil, fmt.Errorf("ImageSets exist but none with the expected name %s", setName)
+	if variantISExists {
+		return nil, fmt.Errorf("ImageSets exist but none with the expected name %s", setName)
+	} else {
+		return nil, nil
+	}
+}
+
+func DoesNonVariantImageSetExist(ctx context.Context, cli client.Client, v operator.ProductVariant) (bool, error) {
```

Comment:
Maybe it can have a comment explaining the requirements. Something like:
- ImageSets are optional by default.
- However, once an ImageSet is specified for your ProductVariant, it is required to have an ImageSet for your ProductVariant and version as well.
- ImageSets for other ProductVariants are not considered for this requirement.